internal/api/users/list.go

```diff
@@ -0,0 +1,39 @@
+package users
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/model"
+)
+
+type ListRequest struct {
+	core.APIRequestUserContext
+	core.MinServerRoleAdmin
+}
+
+type ListResponse struct {
+	Users []*core.ServerUserInfo `json:"users"`
+}
+
+func HandleList(request *ListRequest) (*ListResponse, *core.APIError) {
+	response := ListResponse{}
```

Comment:
Create this closer to where it is used.
(In general (to all programming languages that do not hoist), you want to declare things close to where they are used.)